internal/kafka/internal/config/aws.go

```diff
@@ -19,15 +19,26 @@ type AWSConfig struct {
 	Route53AccessKeyFile       string
 	Route53SecretAccessKey     string
 	Route53SecretAccessKeyFile string
+
+	// Used for AWS Secret Manager
+	SecretManagerAccessKey           string
```

Comment:
Could we put this in its own type? For example, AWSSecretsManagerConfig.
In this way we avoid having the AWSConfig struct with lots of attributes and we can have contained validation and logic for the type

Reply:
I like the suggestion, I'll see what can be done consistently for all the other credentials as well. If not, then I'll defer this change to other PRs

Reply:
done, thanks for the suggestion.